pkg/controllers/dynakube/kspm/daemonset/env.go

```diff
@@ -0,0 +1,65 @@
+package daemonset
+
+import (
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube/kspm"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/capability"
+	agconsts "github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/consts"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	nodeNameEnv           = "KUBERNETES_NODE_NAME"
+	nodeRootEnv           = "KUBERNETES_NODE_ROOT"
+	activeGateEndpointEnv = "DT_ACTIVEGATE_ENDPOINT"
+	tokenEnv              = "DT_K8S_NODE_CONFIGURATION_COLLECTOR_TOKEN"
+)
+
+func getEnvs(dk dynakube.DynaKube, tenantUUID string) []corev1.EnvVar {
+	envs := []corev1.EnvVar{
+		{
+			Name: nodeNameEnv,
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "spec.nodeName",
+				},
+			},
+		},
+		{
+			Name:  nodeRootEnv,
+			Value: nodeRootMountPath,
+		},
+		{
+			Name:  activeGateEndpointEnv,
+			Value: getActiveGateEndpointTemplate(dk, tenantUUID),
+		},
+		{
+			Name: tokenEnv, // TODO: Remove once reading from file is supported
```

Comment:
reading from file should already be supported. you are currently adding the token both ways, so you should be able to remove this env var

Reply:
tried it, with the image I have, and I get an error:
```
Failed contacting ActiveGate: not supported empty Token
```

Either I put the token in the wrong place, or my image doesn't support it yet

Reply:
we are still using the old dev image, so we should try to update that to a newer one. I can help with that if you want

Reply:
done